calico/etcddriver/driver.py

```diff
@@ -512,11 +532,18 @@ def _handle_next_watcher_event(self, resync_in_progress):
         :raises FelixWriteFailed:
         :raises ResyncRequested:
         """
+        # PLW: What happens to the watcher thread if stop() gets called? Or does
+        # it plough on happily until the process goes down?
         if self._watcher_queue is None:
             raise WatcherDied()
         while not self._stop_event.is_set():
             # To make sure we always make progress, only trigger a new resync
             # if we're not in the middle of one.
+            # PLW: This means that when we get a kick to do a resync, we stop
+            # watching for a bit, so could miss an update or two. I think we
+            # should try to close that window, even though it's quite small.
```

Comment:
We could do that, but we could also (for example) remember what index we were at when we told the watcher to die, then use that for the next watcher. That would close the window too.

Or do we already do that? I might be slightly confused here...